lib/source-check.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+const fs = require('fs')
+
+function checkForTranspiledCode (filename) {
+  const readFile = fs.readFileSync(filename, 'utf8')
+  const regex = /function\s+(\w+)/g
+  let matchedObj
+  let isTranspiled = false
+
+  // Check for a source map
+  isTranspiled = readFile.includes('//# sourceMappingURL=')
+
+  while ((matchedObj = regex.exec(readFile)) !== null) {
+    // Avoid infinite loops with zero-width matches
+    if (matchedObj.index === regex.lastIndex) {
+      regex.lastIndex++
+    }
+    // Loop through results and check length of fn name
+    matchedObj.forEach((match, groupIndex) => {
```

Comment:
Do we need to loop here? I think matchedObj is an array-like thing containing `[fullMatch, firstGroup]`, so we could only check for `matchedObj[1]`.

Reply:
Ah you are right. I've removed it since it was overly complicated.